src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -1542,12 +1544,13 @@ internal static unsafe string CharsToStringClass(ReadOnlySpan<char> chars)
             // Get the pointer/length of the span to be able to pass it into string.Create.
 #pragma warning disable CS8500 // takes address of managed type
             ReadOnlySpan<char> tmpChars = chars; // avoid address exposing the span and impacting the other code in the method that uses it
-#if REGEXGENERATOR
-            return StringExtensions.Create(
+            return
+#if NETCOREAPP2_1_OR_GREATER
+                string
 #else
-            return string.Create(
+                StringExtensions
 #endif
-                SetStartIndex + count, (IntPtr)(&tmpChars), static (span, charsPtr) =>
+                .Create(SetStartIndex + count, (IntPtr)(&tmpChars), static (span, charsPtr) =>
```

Comment:
How about you always use `StringExtensions` and add a `global using StringExtensions = System.String;` in frameworks that have `string.Create`? Same with `StackHelper` and `InterlockedExtensions`. This would limit the `#if`s in one place.

I have done something similar in https://github.com/teo-tsirpanis/Recursiont/blob/8ab8331a381c893148c0f21a8fb183cc5d056008/src/Recursiont/Compatibility/RuntimeHelpersCompat.cs.

Reply:
Thanks.  It's a cute trick, but I personally don't like using global usings in that way.  I think it's makes it too hard when looking locally at the code to understand it.